protocol/communities_messenger_token_permissions_test.go

```diff
@@ -959,14 +959,15 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestJoinCommunityAsMemberWit
 			},
 		},
 	}
-	response, err = s.owner.CreateCommunityTokenPermission(&permissionRequestAdmin)
-	s.Require().NoError(err)
-	s.Require().Len(response.Communities(), 1)
 
 	waitOnCommunityPermissionCreated = waitOnCommunitiesEvent(s.owner, func(sub *communities.Subscription) bool {
 		return len(sub.Community.TokenPermissions()) == 2
 	})
 
+	response, err = s.owner.CreateCommunityTokenPermission(&permissionRequestAdmin)
+	s.Require().NoError(err)
+	s.Require().Len(response.Communities(), 1)
+
```

Comment:
I think you didn't change anything. The function call is on line 971.

Reply:
I did :sweat_smile: 

Before:
```
CreateCommunityTokenPermission
waitOnCommunitiesEvent
<-waitOnCommunityPermissionCreated
```


Now:
```
waitOnCommunitiesEvent
CreateCommunityTokenPermission
<-waitOnCommunityPermissionCreated
```

The issue with the former is that `CreateCommunityTokenPermission` can trigger the event before `waitOnCommunitiesEvent` is called. This is causing the flakiness, because sometimes `CreateCommunityTokenPermission` is quick enough to do so. That's why we need to `waitOnCommunitiesEvent` before the action that triggers the event.

Reply:
(That's not a function call, it is waiting on a channel).

Reply:
omg sorry. I misread the function above and thought it was a function being put in a variable that was called below. I just completely misread.

Reply:
No worries, thanks for review :+1: